tests/test_picking_batch.py

```diff
@@ -0,0 +1,307 @@
+        self.create_quant(self.apple.id, self.test_location_01.id, 4,
+                          package_id=other_pack.id)
+        self.create_picking(self.picking_type_pick,
+                            products_info=self.pack_4apples_info,
+                            confirm=True,
+                            assign=True)
+
+        # check pre-conditions
+        self.assertEqual(len(batch.picking_ids), 1)
+        self.assertEqual(batch.state, 'in_progress')
+        self.assertEqual(batch.picking_ids[0].state, 'assigned')
+
+        # method under test
+        with self.assertRaises(ValidationError):
+            Batch.create_batch(None)
+
+    def test11_drop_off_picked(self):
+        """ Marks the batch as done if the picking is complete """
+        Batch = self.env['stock.picking.batch']
+
+        self.create_quant(self.apple.id, self.test_location_01.id, 4,
+                          package_id=self.package_one.id)
+        picking = self.create_picking(self.picking_type_pick,
+                                      products_info=self.pack_4apples_info,
+                                      confirm=True,
+                                      assign=True)
+        batch = Batch.create_batch(None)
+        picking.state = 'done'
+
+        # check pre-conditions
+        self.assertEqual(len(batch.picking_ids), 1)
+        self.assertEqual(batch.state, 'in_progress')
+        self.assertEqual(batch.picking_ids[0].state, 'done')
+
+        # method under test
+        batch.drop_off_picked(True, self.test_location_01.barcode)
+
+        self.assertEqual(batch.state, 'done', "Batch was not completed")
+
+    def _create_valid_batch_for_location_tests(self):
+        Batch = self.env['stock.picking.batch']
+
+        self.create_quant(self.apple.id, self.test_location_01.id, 4,
+                          package_id=self.package_one.id)
+        picking = self.create_picking(self.picking_type_pick,
+                                      products_info=self.pack_4apples_info,
+                                      confirm=True,
+                                      assign=True)
+
+        return picking, Batch.create_batch(None)
+
+    def test12_is_valid_location_dest_success(self):
+        """ Returns True for a valid location """
+        _, batch = self._create_valid_batch_for_location_tests()
+
+        self.assertTrue(
+            batch.is_valid_location_dest_id(self.test_location_02.id),
+            "A valid dest location is wrongly marked as invalid")
+
+    def test13_is_valid_location_dest_failure_invalid_location(self):
+        """ Returns False for a invalid location """
+        Location = self.env['stock.location']
+
+        some_location = Location.create(
+            {'name': "some location name",
+             'barcode': "LTEST13",
+             'location_id': self.test_location_02.id})
+        picking, batch = self._create_valid_batch_for_location_tests()
+        picking.location_dest_id = some_location
```

Comment:
As previous test, if test_location_02 is not a valid location you don't need to do this.